### ISSUE TITLE
configure error, something wrong with OpenSSL?

### ISSUE THREAD

**Issue by iampkuhz** [openlink/virtuoso-opensource#542]
hi, while running `./configure`, it shutdown at

```
checking version and usability of the OpenSSL lib in /usr... configure: error: bad. Check config.log for details
```

while some error in config.log are

```
configure:7094: checking lex library
configure:7108: gcc -o conftest -g -O2   conftest.c   >&5
/tmp/ccKbfYHs.o: In function `input':
/home/hanzhe/Documents/Lab/QA/virtuoso-github-full-backup/lex.yy.c:1183: undefined reference to `yywrap'
/tmp/ccKbfYHs.o: In function `yylex':
/home/hanzhe/Documents/Lab/QA/virtuoso-github-full-backup/lex.yy.c:882: undefined reference to `yywrap'
/tmp/ccKbfYHs.o: In function `main':
/home/hanzhe/Documents/Lab/QA/virtuoso-github-full-backup/conftest.l:18: undefined reference to `yywrap'
collect2: error: ld returned 1 exit status
configure:7108: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "Virtuoso Open Source Edition (Column Store)"
| #define PACKAGE_TARNAME "virtuoso-opensource"
```

and

```
/home/hanzhe/Documents/Lab/QA/virtuoso-github-full-backup/conftest.c:82: undefined reference to `t_open'
collect2: error: ld returned 1 exit status
configure:14996: $? = 1
configure: failed program was:
```

and so on..

i think it's something wrong with my openssl?
because the `configure:16495: error: bad. Check config.log for details` lie in

```
/home/hanzhe/Documents/Lab/QA/virtuoso-github-full-backup/conftest.c:163: undefined reference to `OPENSSL_init_ssl'
/home/hanzhe/Documents/Lab/QA/virtuoso-github-full-backup/conftest.c:170: undefined reference to `OPENSSL_init_ssl'
/home/hanzhe/Documents/Lab/QA/virtuoso-github-full-backup/conftest.c:173: undefined reference to `TLS_client_method'
collect2: error: ld returned 1 exit status
configure:16481: $? = 1
configure: program exited with status 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "Virtuoso Open Source Edition (Column Store)"
| #define PACKAGE_TARNAME "virtuoso-opensource"
| #define PACKAGE_VERSION "7.2.2"
| #define PACKAGE_STRING "Virtuoso Open Source Edition (Column Store) 7.2.2"
| #define PACKAGE_BUGREPORT "vos.admin@openlinksw.com"
| #define PACKAGE_URL ""
| #define PACKAGE "virtuoso-opensource"
| #define VERSION "7.2.2"
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define __EXTENSIONS__ 1
| #define _ALL_SOURCE 1
| #define _GNU_SOURCE 1
| #define _POSIX_PTHREAD_SEMANTICS 1
| #define _TANDEM_SOURCE 1
| #define HAVE_LONG_DOUBLE_WIDER 1
| #define HAVE_LONG_DOUBLE 1
| #define SIZEOF_CHAR 1
| #define SIZEOF_SHORT 2
| #define SIZEOF_UNSIGNED 4
| #define SIZEOF_INT 4
| #define SIZEOF_LONG 8
| #define SIZEOF_LONG_LONG 8
| #define SIZEOF_FLOAT 4
| #define SIZEOF_DOUBLE 8
| #define SIZEOF_LONG_DOUBLE 16
| #define SIZEOF_CHAR_P 8
| #define SIZEOF_VOID_P 8
| #define YYTEXT_POINTER 1
| #define HAVE_DLFCN_H 1
| #define LT_OBJDIR ".libs/"
| #define STDC_HEADERS 1
| #define HAVE_DIRENT_H 1
| #define TIME_WITH_SYS_TIME 1
| #define HAVE_SYS_WAIT_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_LIMITS_H 1
| #define HAVE_SYS_PARAM_H 1
| #define HAVE_FCNTL_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_SYS_TIMEB_H 1
| #define HAVE_SYS_RESOURCE_H 1
| #define HAVE_MALLOC_H 1
| #define HAVE_SYS_SELECT_H 1
| #define HAVE_SYS_TIME_H 1
| #define HAVE_WCHAR_H 1
| #define HAVE_WCTYPE_H 1
| #define HAVE_PWD_H 1
| #define HAVE_GRP_H 1
| #define HAVE_SYS_MMAN_H 1
| #define RETSIGTYPE void
| #define HAVE_STRUCT_STAT_ST_BLKSIZE 1
| #define HAVE_ST_BLKSIZE 1
| #define HAVE_STRUCT_STAT_ST_BLOCKS 1
| #define HAVE_ST_BLOCKS 1
| #define HAVE_STRUCT_STAT_ST_RDEV 1
| #define HAVE_ST_RDEV 1
| #define HAVE_PTRDIFF_T 1
| #define HAVE_TM_GMTOFF 1
| #define HAVE_FORK 1
| #define HAVE_VFORK 1
| #define HAVE_WORKING_VFORK 1
| #define HAVE_WORKING_FORK 1
| #define HAVE_UNISTD_H 1
| #define HAVE_CHOWN 1
| #define HAVE_DECL_STRERROR_R 1
| #define HAVE_STRERROR_R 1
| #define STRERROR_R_CHAR_P 1
| #define HAVE_GETPAGESIZE 1
| #define HAVE_GETTIMEOFDAY 1
| #define HAVE_STRTOK_R 1
| #define HAVE_FTRUNCATE 1
| #define HAVE_FCHMOD 1
| #define HAVE_FSYNC 1
| #define HAVE_FTIME 1
| #define HAVE_LOCALTIME_R 1
| #define HAVE_GMTIME_R 1
| #define HAVE_SYSLOG 1
| #define HAVE_SNPRINTF 1
| #define HAVE_VSNPRINTF 1
| #define HAVE_WCSDUP 1
| #define HAVE_WCSCASECMP 1
| #define HAVE_WCSCMP 1
| #define HAVE_WCSLEN 1
| #define HAVE_WCSNCAT 1
| #define HAVE_WCSNCASECMP 1
| #define HAVE_WCSNCMP 1
| #define HAVE_WCSCPY 1
| #define HAVE_ISWSPACE 1
| #define HAVE_ISWDIGIT 1
| #define HAVE_WCSNCPY 1
| #define HAVE_WCSCHR 1
| #define HAVE_WCSRCHR 1
| #define HAVE_WCSSTR 1
| #define HAVE_WCSCAT 1
| #define HAVE_ISWLOWER 1
| #define HAVE_ISWUPPER 1
| #define HAVE_TOWLOWER 1
| #define HAVE_TOWUPPER 1
| #define HAVE_WPRINTF 1
| #define HAVE_FPUTWS 1
| #define HAVE_PUTWC 1
| #define HAVE_SWPRINTF 1
| #define HAVE_FWPRINTF 1
| #define HAVE_SWSCANF 1
| #define HAVE_FGETWS 1
| #define HAVE_PUTWCHAR 1
| #define HAVE_GETPWNAM 1
| #define HAVE_GETGRNAM 1
| #define HAVE_CHMOD 1
| #define HAVE_GETPWUID 1
| #define HAVE_GETGRGID 1
| #define HAVE_SETUID 1
| #define HAVE_PUTENV 1
| #define HAVE_SETENV 1
| #define HAVE_STRERROR 1
| #define HAVE_STRDUP 1
| #define HAVE_GETRUSAGE 1
| #define HAVE_MEMMOVE 1
| #define HAVE_MEMMEM 1
| #define HAVE_MEMCPY 1
| #define HAVE_STRFTIME 1
| #define HAVE_STPCPY 1
| #define HAVE_CLOCK_GETTIME 1
| #define HAVE_SYS_ERRLIST 1
| #define HAVE_VA_COPY 1
| #define HAVE___VA_COPY 1
| #define HAVE_FLOCK_IN_SYS_FILE 1
| #define HAVE_LONG_FILE_NAMES 1
| #define HAVE_RESTARTABLE_SYSCALLS 1
| #define BIF_XML 1
| /* end confdefs.h.  */
| 
| #include <openssl/rsa.h>
| #include <openssl/crypto.h>
| #include <openssl/x509.h>
| #include <openssl/pem.h>
| #include <openssl/ssl.h>
| #include <openssl/err.h>
| #include <openssl/bio.h>
| #include <openssl/asn1.h>
| #include <openssl/pkcs12.h>
| #include <openssl/rand.h>
| #include <stdio.h>
| #include <string.h>
| int
| main ()
| {
|     SSL_METHOD *ssl_server_method = NULL;
|     SSL_CTX *ssl_server_ctx = NULL;
|     unsigned char tmp[1024];
| 
|     SSL_load_error_strings();
|     ERR_load_X509_strings();
|     ERR_load_PKCS12_strings();
|     #ifndef WIN32
|     RAND_bytes (tmp, sizeof(tmp));
|     RAND_add (tmp, sizeof(tmp), (double)(sizeof(tmp)));
|     #endif
|     SSLeay_add_ssl_algorithms();
|     PKCS12_PBE_add ();
| 
|     ssl_server_method = SSLv23_client_method();
|     ssl_server_ctx = SSL_CTX_new (ssl_server_method);
| 
|     if (!ssl_server_ctx)
|     {
|   ERR_print_errors_fp(stderr);
|   return(-1);
|     }
| 
|     if (OPENSSL_VERSION_NUMBER < 0x0090700f)
|     {
|   fprintf (stderr, "Wrong SSL version number\n");
|   return (-1);
|     }
|   return 0;
| }
configure:16495: error: bad. Check config.log for details
```

btw, the output ot `./autogen` print my openssl version is `Using OpenSSL 1.1.0-pre4-dev  xx XXX xxxx`

I have struggling with this issue for many days.... 

Could anyone give me some help?

**Comment by openlink** [comment]
On 17 March 2016 at 13:27, iampkuhz notifications@github.com wrote:

> configure:7094: checking lex library configure:7108: gcc -o conftest -g
> -O2 conftest.c >&5 /tmp/ccKbfYHs.o: In function `input':
> /home/hanzhe/Documents/Lab/QA/virtuoso-github-full-backup/lex.yy.c:1183:
> undefined reference to`yywrap' /tmp/ccKbfYHs.o: In function `yylex':

Hi,

This looks more like you don't have an appropriate version of flex
installed. Please can you check the build requirements at
http://virtuoso.openlinksw.com/dataspace/doc/dav/wiki/Main/VOSMake#Requirements
and see if something's lacking?

Regards,

## ~Tim

Tim Haynes
Product Development Consultant
OpenLink Software
http://www.openlinksw.com/
http://twitter.com/openlink

**Comment by iampkuhz** [comment]
glad to see this reply!! T_T

flex is OK ? 
run `flex --version` response `flex 2.5.35`

**Comment by openlink** [comment]
You have the `openssl` runtime and utilities installed, as this is checked by the `autogen.sh` script; but you may be missing the `openssl` development libraries which are checked by configure. This package maybe called `openssl-devel`, `libopenssl-dev`, or similar. 

What operating system and version are you currently using, so i can check the actual name for you?

**Comment by TallTed** [comment]
@iampkuhz - Just in case this is still a problem for you, please see the updated [README.OpenSSL.md](https://github.com/openlink/virtuoso-opensource/blob/fix/TallTed-20190723-README.OpenSSL.md/README.OpenSSL.md) and [README](https://github.com/openlink/virtuoso-opensource/blob/fix/TallTed-20190723-README-cleanup/README). In any case, please let us know if we can consider this issue closed.
